src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse.cs

```diff
@@ -16,373 +16,490 @@ public static class Sse
+
+        /// <summary>
+        /// void _mm_prefetch(char* p, int i);
+        ///   PREFETCHNTA m8
+        /// </summary>
+        public static unsafe void PrefetchNonTemporal(byte* address) => PrefetchNonTemporal(address);
+
+        /// <summary>
+        /// __m128 _mm_rcp_ps (__m128 a);
+        ///   RCPPS xmm, xmm/m128
         /// </summary>
         public static Vector128<float> Reciprocal(Vector128<float> value) => Reciprocal(value);
 
         /// <summary>
-        /// __m128 _mm_rcp_ss (__m128 a)
+        /// __m128 _mm_rcp_ss (__m128 a);
+        ///   RCPSS xmm, xmm/m32
         /// </summary>
         public static Vector128<float> ReciprocalScalar(Vector128<float> value) => ReciprocalScalar(value);
 
         /// <summary>
-        /// __m128 _mm_rsqrt_ps (__m128 a)
+        /// __m128 _mm_rcp_ss (__m128 a, __m128 b);
+        ///   RCPSS xmm, xmm/m32
+        /// The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.
+        /// </summary>
+        public static Vector128<float> ReciprocalScalar(Vector128<float> upper, Vector128<float> value) => ReciprocalScalar(upper, value);
+
+        /// <summary>
+        /// __m128 _mm_rsqrt_ps (__m128 a);
+        ///   RSQRTPS xmm, xmm/m128
         /// </summary>
         public static Vector128<float> ReciprocalSqrt(Vector128<float> value) => ReciprocalSqrt(value);
 
         /// <summary>
-        /// __m128 _mm_rsqrt_ss (__m128 a)
+        /// __m128 _mm_rsqrt_ss (__m128 a);
+        ///   RSQRTSS xmm, xmm/m32
         /// </summary>
         public static Vector128<float> ReciprocalSqrtScalar(Vector128<float> value) => ReciprocalSqrtScalar(value);
 
+        /// <summary>
+        /// __m128 _mm_rsqrt_ss (__m128 a, __m128 b);
+        ///   RSQRTSS xmm, xmm/m32
+        /// The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.
+        /// </summary>
+        public static Vector128<float> ReciprocalSqrtScalar(Vector128<float> upper, Vector128<float> value) => ReciprocalSqrtScalar(upper, value);
+
         /// <summary>
         /// __m128 _mm_set_ps (float e3, float e2, float e1, float e0)
         /// </summary>
         public static Vector128<float> SetVector128(float e3, float e2, float e1, float e0) => SetVector128(e3, e2, e1, e0);
 
         /// <summary>
-        /// __m128 _mm_set_ss (float a)
+        /// __m128 _mm_set_ss (float a);
+        ///   HELPER
         /// </summary>
-        public static Vector128<float> SetScalar(float value) => SetScalar(value);
+        public static Vector128<float> SetScalarVector128(float value) => SetScalarVector128(value);
 
         /// <summary>
-        /// __m128 _mm_set1_ps (float a)
+        /// __m128 _mm_set1_ps (float a);
+        ///   HELPER
         /// </summary>
         public static Vector128<float> SetAllVector128(float value) => SetAllVector128(value);
```

Comment:
It's not directly related to this commit but I forget to ask it - perhaps we should generate `(V)BROADCAST` instructions when on AVX/AVX2

Reply:
Yes. Helper instructions don't make a guarantee about a particular instruction being emitted, they actually make a guarantee about behavior instead.